### DISCUSSION TITLE
Restart a simulation

### DISCUSSION THREAD

**Discussion by bjmvandongen** [firemodels/fds#13643] [Q&A]
A simulation that I ran on a cluster stopped because my disk quota was exceeded. 
I want to restart this simulation. 
I have added RESTART=T on the MISC line and I have added the CHID.restart just as is described in the user guide.
I use 32 processors for this simulation. I get this error for every processor:

CHID_1.restart does not exist in the current directory. 
...
CHID_32.restart does not exist in the current directory. 

So I have added these files to the folder, but then I got the error:
forrtl: severe (24): end-of-file during read, unit 130, file and then the directory to the folder and the file CHID_1.restart. 

Does somebody know if I can restart this simulation or what I'm doing wrong?

**Comment by rmcdermo** [comment]
If you did not ask FDS to write a restart file in your initial simulation, then it will not be there.  Search RESTART in the user guide.

**Comment by bjmvandongen** [comment]
Thank you for your answer.
I see now that the restart file is automatically created when you stop the simulation. 
In my case, the simulation stopped due to a disk quota that was exceeded and thus no restart file was made. I will run the simulation from the beginning again.

**Comment by rmcdermo** [comment]
A long time ago we used to write plot3d at a default interval.  We never did this for RESTART because historically it has been considered fragile.  But since the implementation of restart cases in our verification suite, I think this has become more robust, and I would be willing to consider a default restart interval.

@drjfloyd @lu-kas @shostikk @marcosvanella @ericvmueller @mcgratta what are your thoughts on this?

**Reply by ericvmueller** [reply]
The only thing is the increase in required disk storage size ...but maybe that is a non-issue for most people.

**Reply by drjfloyd** [reply]
No real opposition to this. I wouldn't want it to be very frequent as restart files take a while to write.  
DT_RESTART=(T_END-T_BEGIN)*0.2 _EB or 0.25_EB?

If we write these by default we may want to change all the verification cases to not write them (except for cases testing restart) so that we don't blow up the time needed.

**Reply by marcosvanella** [reply]
Would we be overwriting the restart files dumped on the previous DT_RESTART or we want to keep more than one set of restart files? The only reason for the latter would be the highly unusual case that for whatever reason the restart writing gets corrupted (i.e. exceeding disk quota when writing the restart files).

**Reply by rmcdermo** [reply]
@drjfloyd Yes something like defaulting to 5 restart writes per simulation was what we were thinking.

@marcosvanella For now we were just thinking of setting DT_RESTART by default, which would overwrite at the default interval.

Of course, the user can always set DT_DEFAULT to their desired value.

**Reply by drjfloyd** [reply]
I would not want the default behavior to be new files everytime they are written.  That could chew up and incredicble amount of disk space for very large jobs.
